lib/puppet/rest/compression.rb

```diff
@@ -0,0 +1,31 @@
+module Puppet::Rest
+  class Uncompressor
+
+    def initialize
+      @zlib = Puppet::Network::HTTP::Compression::Active::ZlibAdapter.new
+    end
+
+    # Closes the compression stream.
+    def self.close
+      @zlib.close
+    end
+
+    # Process the current chunk based on the given content encoding.
+    # If the chunk is compressed, uncompress it, otherwise return it as is.
+    # @param [String] content_encoding the encoding of the chunk to process
+    # @param [String] chunk a section of the response being processed
+    # @raise [Puppet::Rest::ResponseError] if passed an unknown content encoding.
+    # @return [String] the processed chunk
+    def uncompress_chunk(content_encoding, chunk)
+      case content_encoding
```

Comment:
Looking up more info about the client I found this blog post:
[httpclient is a nice http client](https://bibwild.wordpress.com/2012/04/30/httpclient-is-a-nice-http-client-forin-ruby/).

Which mentions that it can do transparent gzip decompression for us. Have you tried configuring [this](https://github.com/nahi/httpclient/blob/v3.2.8/lib/httpclient.rb#L372) (sorry if this already came up)?

Reply:
Oh, no, I hadn't come across that it could do that. I'll look into that, seems nice! I'll have to look into making sure that works with the streaming API.

Reply:
Totally works. Excellent.